benchmarks/benchmarks.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+require "benchmark/ips"
+
+require "memery"
+require "memo_wise"
+require "memoist"
+require "memoized"
+require "memoizer"
+
+BenchmarkGem = Struct.new(:klass, :inheritance_method, :memoization_method)
+
+# We alphabetize this list for easier readability, but shuffle the list before
+# using it to minimize the chance that our benchmarks are affected by ordering.
+BENCHMARK_GEMS = [
+  BenchmarkGem.new(Memery, :include, :memoize),
+  BenchmarkGem.new(MemoWise, :prepend, :memo_wise),
+  BenchmarkGem.new(Memoist, :extend, :memoize),
+  BenchmarkGem.new(Memoized, :include, :memoize),
+  BenchmarkGem.new(Memoizer, :include, :memoize)
+].shuffle
+
+# Use metaprogramming to ensure that each class is created in exactly the
```

Comment:
I don't love doing so much metaprogramming here, but without this the file got really big and I worried about subtle errors (e.g. accidentally copy-pasting something wrong and testing the wrong gem). Thoughts?

Reply:
Really I just worry that the way I'm adding metaprogramming will add confounding factors that influence the results. Would appreciate review with an eye toward "yes this seems okay" or "no this will invalidate our results."

Reply:
Talked over zoom, I think we addressed any potential confounding factors.